Lib/urllib/parse.py

```diff
@@ -942,20 +942,17 @@ def splittype(url):
         return scheme.lower(), data
     return None, url
 
-_hostprog = None
 def splithost(url):
     """splithost('//host[:port]/path') --> 'host[:port]', '/path'."""
-    global _hostprog
-    if _hostprog is None:
-        _hostprog = re.compile('//([^/?]*)(.*)', re.DOTALL)
-
-    match = _hostprog.match(url)
-    if match:
-        host_port, path = match.groups()
-        if path and path[0] != '/':
-            path = '/' + path
-        return host_port, path
-    return None, url
+    fields = urlparse(url)
```

Comment:
@postmasters rewrote this change to only modify the regex, so it doesn't change the behaviour anymore.